### PR TITLE
check_sql: allow configuration of metric name

### PR DESCRIPTION
**Pull request by gurubert**
This patch adds an optional argument to --metrics that will be used as metric name for the perfomance data.
If no argument is passed, "performance_data" is still the default metric name.

### CODE REVIEW

**Comment by jonaskluger** [comment]
Hello Robert,

thank you for this request. We will have a closer look at your patch and discuss internally on how and if we can implement it into the official code base.

Internal Ref: CMK-3202

**Comment by auchri** [comment]
Hi @jherbel, it would be awesome if you could merge this!

File: cmk/gui/plugins/wato/active_checks.py [hunk 1]
```diff
@@ -545,6 +545,10 @@ def _valuespec_active_checks_dns():
         valuespec=_valuespec_active_checks_dns,
     ))
 
+def transform_check_sql_perfdata(perfdata):
+    if not isinstance(perfdata, str):
+        return "performance_data"
+    return perfdata
```
Comment (mo-ki) [inline cmk/gui/plugins/wato/active_checks.py:551]: If you swap those around, you can loose the 'not'

File: cmk/gui/plugins/wato/active_checks.py [hunk 2]
```diff
@@ -660,10 +664,13 @@ def _valuespec_active_checks_sql():
                    elements=[Float(title=_("Warning below")),
                              Float(title=_("Critical below"))])),
             ("perfdata",
-             FixedValue(
-                 title=_("Performance Data"),
-                 totext=_("Store output value into RRD database"),
-                 value=True,
+             Transform(
+                 TextAscii(
```
Comment (mo-ki) [inline cmk/gui/plugins/wato/active_checks.py:668]: TextInput  (Ascii vs. Unicode is legacy)

File: cmk/gui/plugins/wato/active_checks.py [hunk 3]
```diff
@@ -660,10 +664,13 @@ def _valuespec_active_checks_sql():
...
+                     title=_("Performance Data"),
+                     help=_("Store output value into RRD database in a metric with this name."),
+                     default_value='performance_data',
+                 ),
```
Comment (mo-ki) [inline cmk/gui/plugins/wato/active_checks.py:672]: I think we should also have `allow_empty=False`.

File: checks/check_sql [hunk 4]
```diff
@@ -38,7 +38,9 @@ def check_sql_arguments(params):
         lower = "", ""
 
     if "perfdata" in params:
-        args.append("--metrics")
+        if not params["perfdata"]:
+            params["perfdata"] = "performance_data"
+        args.append("--metrics=%s" % params["perfdata"])
```
Comment (mo-ki) [inline checks/check_sql:43]: There are some issues here.
I don't think we can have the `not params["perfdata"]` case, if we prohibit empty strings (see below).
Changing the functions argument (params) is dangerous. I suggest
```
if (metrics:=params.get("perfdata")) is not None:
    args.append(f"--metrics={metrics}")
```

Please add a test for this in tests/unit/checks/test_check_sql.py!

**Review by mo-ki — CHANGES_REQUESTED** [review]
Hi Robert!
If you can address the mentioned issues, I think we should mere this. Looks like a straightforward improvement to me.

**Comment by mo-ki** [comment]
Believe it or not, I am about to merge this.

**Comment by gurubert** [comment]
😱😊

**Pull request closed without merging**